### ISSUE TITLE
Error in resolving json module

### ISSUE THREAD

**Issue by EmCeeEs** [LeDDGroup/typescript-transform-paths#95]
**Description**
Using `"resolveJsonModule": true` in `tsconfig.json`  results in import error after compilation.
The reason is that the `.json` extension is removed.

**Expected Behavior**
`import data from  './src/data.json'` => `import data from  './src/data.json'`
`import data from '@src/data.json'` => `import data from './src/data.json'`

**Actual Behavior**
`import data from './src/data.json'` => `import data from './src/data'`
`import data from '@src/data.json'` => `import data from './src/data'`

**Version**
├─ ttypescript@1.5.12
├─ typescript-transform-paths@2.2.2
└─ typescript@4.1.3

**Config**
```
{
  "compilerOptions": {
    "target": "es5",
    "module": "commonjs",
    "strict": true,
    "esModuleInterop": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
    "resolveJsonModule": true,
    "baseUrl": ".",
    "paths": {
      "@*": ["src/*"]
    },
    "plugins": [
      { "transform": "typescript-transform-paths" },
      { "transform": "typescript-transform-paths", "afterDeclarations": true }
    ]
  },
  "files": ["src/index.ts"]
}
```

**Comment by EmCeeEs** [comment]
Quick fix by using `@no-transform-path` tag

**Comment by nonara** [comment]
Thanks for the report! 

@danielpza The logic for removing the `json` extension was a bit before my time. I don't want to remove it altogether without knowing the full context. Do you have any insight on why it was put in and if it used to work with TS?

**Comment by danielpza** [comment]
I think it was for resolving json files without the .json extension, like require('./file') would require './file.json'. I'm not sure how valid is that tbh.

**Comment by nonara** [comment]
Ok. I'll have a look over the require specs and compiler and see. I'm not sure about that either.

@EmCeeEs I'm out for the rest of this week but will try to get it wrapped next week! In the mean time you can use the `@transform-path` tag to manually transform.

**Comment by nonara** [comment]
### Update

I had a look at it. I think the potential issues here are broader than even the `json` case. In general, we should not be adding or removing file extensions from how they are specified in the source.

I'm going to change the logic for removing extensions to essentially mirror the input format. That should cover this as well as any other potential issues. I expect to have this wrapped up tomorrow.

**Comment by nonara** [comment]
Fixed in v2.2.3
